Client/Frontend/Settings/NotificationsSettingsViewController.swift

```diff
@@ -0,0 +1,179 @@
+                    self.engagementNotificationHelper.schedule()
+                } else {
+                    // cancel all pending engagement notifications
+                    self.engagementNotificationHelper.cancelAll()
+                }
+            }
+        }
+    }()
+
+    private let prefs: Prefs
+    private let hasAccount: Bool
+    private var footerTitle = ""
+
+    init(prefs: Prefs, hasAccount: Bool) {
+        self.prefs = prefs
+        self.hasAccount = hasAccount
+        super.init(style: .grouped)
+        self.title = .Settings.Notifications.Title
+        self.addObservers()
+
+        Task {
+            await self.checkForSystemNotifications()
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func generateSettings() -> [SettingSection] {
+        let childrenSection: [Setting]
+        if hasAccount {
+            childrenSection = [syncNotifications, tipsAndFeaturesNotifications]
+        } else {
+            childrenSection = [tipsAndFeaturesNotifications]
+        }
+
+        return [
+            SettingSection(footerTitle: NSAttributedString(string: footerTitle), children: childrenSection)
+        ]
+    }
+
+    func checkForSystemNotifications() async {
+        let settings = await NotificationManager().getNotificationSettings()
+
+        switch settings.authorizationStatus {
+        case .authorized, .provisional, .ephemeral:
+            self.footerTitle = ""
+        case .denied:
+            self.footerTitle = String(format: .Settings.Notifications.systemNotificationsDisabledMessage, AppName.shortName.rawValue, AppName.shortName.rawValue)
+        case .notDetermined:
+            self.footerTitle = ""
+        @unknown default:
+            self.footerTitle = ""
+        }
+
+        self.settings = generateSettings()
+        self.tableView.reloadData()
+    }
+
+    private func notificationsChanged(_ sendNotifications: Bool) async -> Bool {
+        guard sendNotifications else { return false }
+
+        let notificationManager = NotificationManager()
+        let settings = await notificationManager.getNotificationSettings()
+
+        switch settings.authorizationStatus {
+        case .authorized, .provisional, .ephemeral:
+            return true
+        case .notDetermined:
+            do {
+                return try await notificationManager.requestAuthorization()
+            } catch {
+                return false
+            }
+        case .denied:
+            self.footerTitle = String(format: .Settings.Notifications.systemNotificationsDisabledMessage, AppName.shortName.rawValue, AppName.shortName.rawValue)
+            self.settings = generateSettings()
+            self.tableView.reloadData()
+            await MainActor.run {
+                self.present(accessDeniedAlert, animated: true, completion: nil)
+            }
+            return false
```

Comment:
Question: how does this happen, for multi-thread purposes? `present` goes before the `return` always, due to the `await` right? I genuinely don't know.

Reply:
So `await` is a suspension point but it does not block the execution of the code. The method will most likely return, but the presentation will suspend util some time is available to run on main. Technically it should be instant, but if the main thread is for some reason blocked it will have to wait so that's why it's market as awaiting. Kinda similar with escaping closure.